internal/controller/ovn_ipfix_collector.go

```diff
@@ -0,0 +1,288 @@
+							if err != nil {
+								logger.Error("Failed to create layer4 message", zap.Error(err))
+								continue
+							}
+							convertOvnFlow := &pb.StandardFlow{
+								Layer3: layer3Message,
+								Layer4: layer4Message,
+								Ts: &pb.StandardFlow_Timestamp{
+									Timestamp: timestamppb.New(time.Now()),
+								},
+							}
+							err = sm.FlowCache.CacheFlow(ctx, convertOvnFlow)
+							if err != nil {
+								logger.Error("Failed to cache flow", zap.Error(err))
+								return err
+							}
+						}
+					}
+				}
+			} else {
+				logger.Info("Not an IPFIX packet")
+			}
+		}
+	}
+}
+
+// parseIPv4Address converts a byte slice into an IPv4 address string.
+// Returns an error if the slice is not the correct size.
+func parseIPv4Address(decodedValue []byte) (string, error) {
+	if len(decodedValue) < 4 {
+		return "", errors.New("insufficient data to parse IPv4 address")
+	}
+	ip := net.IPv4(decodedValue[0], decodedValue[1], decodedValue[2], decodedValue[3])
+	return ip.String(), nil
+}
+
+// parsePort converts a byte slice into a uint16 port number using BigEndian encoding.
+// Returns an error if the slice is not the correct size.
+func parsePort(decodedValue []byte) (uint16, error) {
+	if len(decodedValue) < 2 {
+		return 0, errors.New("insufficient data to parse port")
+	}
+	return binary.BigEndian.Uint16(decodedValue), nil
+}
+
+// parseProtocol converts a byte slice into a protocol string based on IANA protocol numbers.
+// Returns an error if the slice is not the correct size or the protocol is unknown.
+func parseProtocol(decodedValue []byte) (string, error) {
+	if len(decodedValue) < 1 {
+		return "", errors.New("insufficient data to parse protocol")
+	}
+
+	protocol := uint8(decodedValue[0])
+	switch protocol {
+	case 1:
+		return "icmpt", nil
+	case 6:
+		return "tcp", nil
+	case 17:
+		return "udp", nil
+	case 132:
+		return "sctp", nil
+	default:
+		return "", errors.New("unknown protocol")
+	}
+}
+
+// parseIPVersion converts a byte slice into an IP version string (e.g., "ipv4" or "ipv6").
+// Returns an error if the slice is not the correct size or the IP version is unknown.
+func parseIPVersion(decodedValue []byte) (string, error) {
+	if len(decodedValue) < 1 {
+		return "", errors.New("insufficient data to parse IP version")
+	}
+
+	ipVersion := uint8(decodedValue[0])
+	switch ipVersion {
+	case 4:
+		return "ipv4", nil
+	case 6:
+		return "ipv6", nil
+	default:
+		return "", errors.New("unknown IP version")
+	}
+}
+
+// processDataRecord processes a single data record and converts it into an OVNFlow.
+// If any parsing step fails, it returns an error and skips the record.
+func processDataRecord(dataRecord netflows.DataRecord) (OVNFlow, error) {
+	ovnFlow := OVNFlow{}
+	for _, field := range dataRecord.Values {
+		switch field.Type {
+		case 8: // Assuming 8 corresponds to sourceIPv4Address
```

Comment:
Yes, you're assuming. No need to comment about that. :)
Those field types are standard and officially defined by the IANA.
```suggestion
		case 8: // sourceIPv4Address
```
Same for the other comments below.